### PR TITLE
HADOOP-18351. Reduce excess logging of errors during S3A prefetching reads

### PR DESCRIPTION
**Pull request by sauraank**
### Description of PR
Logged the errors at the info level with just the message. Added the stack at the Debug level. Not made any changes where error stack was not present and had only the message and parameters at Warn/INFO level.
Also, removed the case of multiple logging in case of prefetch. Moved the log one method above ReadBlock() (which was common for sync read thread and async prefetch thread) and logged it at ERROR level for the case of read.

### How was this patch tested?
Logging levels of Error/info was tested locally by throwing errors in case of both cases of async prefetch and sync get calls for read separately. Then the unit test of ITestS3APrefetchingInputStream integration test ( testReadLargeFileFully) was used.
Finally done the integration test by `mvn -Dparallel-tests clean verify` in the `eu-west-1` region.
Following is the result of the tests - 
[INFO] Results:
[INFO]
[WARNING] Tests run: 1154, Failures: 0, Errors: 0, Skipped: 182
[INFO]
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M1:integration-test (sequential-integration-tests) @ hadoop-aws ---
[INFO] Results:
[INFO]
[WARNING] Tests run: 124, Failures: 0, Errors: 0, Skipped: 84
[INFO]
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (depcheck) @ hadoop-aws ---
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M1:verify (default-integration-test) @ hadoop-aws ---
[INFO]
[INFO] --- maven-failsafe-plugin:3.0.0-M1:verify (sequential-integration-tests) @ hadoop-aws ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 14:05 min
[INFO] Finished at: 2023-02-14T11:25:12Z
[INFO] Final Memory: 76M/1674M
[INFO] ------------------------------------------------------------------------

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 49s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 37s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  28m 11s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  25m 47s |  |  trunk passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  compile  |  22m 16s |  |  trunk passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   4m  2s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 28s |  |  trunk passed  |
| -1 :x: |  javadoc  |   1m  8s | [/branch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/1/artifact/out/branch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt) |  hadoop-common in trunk failed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.  |
| +1 :green_heart: |  javadoc  |   1m 18s |  |  trunk passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   3m 54s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  23m 47s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 32s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  24m 43s |  |  the patch passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  javac  |  24m 43s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  21m 51s |  |  the patch passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |  21m 51s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   3m 53s | [/results-checkstyle-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/1/artifact/out/results-checkstyle-root.txt) |  root: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| +1 :green_heart: |  mvnsite  |   2m 24s |  |  the patch passed  |
| -1 :x: |  javadoc  |   1m  0s | [/patch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/1/artifact/out/patch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt) |  hadoop-common in the patch failed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  |  the patch passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   4m  7s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  24m  5s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  18m 25s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   3m  1s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 54s |  |  The patch does not generate ASF License warnings.  |
|  |   | 239m 44s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5274 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 465c3f60a6e5 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 6c81a1f8909d6f68b5a39e2aef0dbe838fe58fb6 |
| Default Java | Private Build-1.8.0_352-8u352-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_352-8u352-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/1/testReport/ |
| Max. process+thread count | 2305 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 49s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 13s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  34m  0s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  25m 19s |  |  trunk passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  compile  |  21m 41s |  |  trunk passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   3m 59s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 27s |  |  trunk passed  |
| -1 :x: |  javadoc  |   1m  9s | [/branch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/2/artifact/out/branch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt) |  hadoop-common in trunk failed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.  |
| +1 :green_heart: |  javadoc  |   1m 19s |  |  trunk passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   3m 54s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  26m 38s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 51s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  24m 31s |  |  the patch passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  javac  |  24m 31s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  21m 37s |  |  the patch passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |  21m 37s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   3m 53s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   2m 23s |  |  the patch passed  |
| -1 :x: |  javadoc  |   0m 59s | [/patch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/2/artifact/out/patch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.txt) |  hadoop-common in the patch failed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04.  |
| +1 :green_heart: |  javadoc  |   1m 18s |  |  the patch passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   4m  3s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  28m  6s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  19m 21s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 50s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 52s |  |  The patch does not generate ASF License warnings.  |
|  |   | 252m 50s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5274 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 9b24f24902f0 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 2dd89ea3532590fc15218b4554bb85ef037e97ef |
| Default Java | Private Build-1.8.0_352-8u352-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_352-8u352-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/2/testReport/ |
| Max. process+thread count | 3144 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by sauraank** [comment]
Hey @steveloughran ! Can you please review it. Tested the logging of Errors by throwing errors in case of both prefetch and read.
Thanks

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/CachingBlockManager.java [hunk 1]
```diff
@@ -302,7 +302,13 @@ public void cancelPrefetches() {
 
   private void read(BufferData data) throws IOException {
     synchronized (data) {
-      readBlock(data, false, BufferData.State.BLANK);
+      try {
+        readBlock(data, false, BufferData.State.BLANK);
+      } catch(Exception e) {
```
Comment (steveloughran) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/CachingBlockManager.java]: add a space between catch and "(", and only catch IOException

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/CachingBlockManager.java [hunk 2]
```diff
@@ -302,7 +302,13 @@ public void cancelPrefetches() {
...
+        String message = String.format("error reading block %s", data.getBlockNumber());
+        LOG.error(message, e);
```
Comment (steveloughran) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/CachingBlockManager.java]: the string formatting can go into the LOG.error() just like the ones below

**Review by steveloughran — CHANGES_REQUESTED** [review]
looks good, some minor nits to deal with and then it is good to go in.

It'll be interesting to see what happens in real use when we encounter errors here. With the prefetching things may surface sooner, but transient ones may now be recovered from better.

**Comment by sauraank** [comment]
Hey @steveloughran ! Made the small changes recommended by you and ran the tests again. Thanks. Please review it.

**Review by steveloughran — APPROVED** [review]
LGTM
+1

**Comment by steveloughran** [comment]
will merge as soon as the yetus build is happy

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 51s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m  6s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  33m 41s |  |  trunk passed  |
| +1 :green_heart: |  compile  |  25m 24s |  |  trunk passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  compile  |  21m 48s |  |  trunk passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  checkstyle  |   4m  1s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 30s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 37s |  |  trunk passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  javadoc  |   1m 20s |  |  trunk passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   3m 53s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  26m 37s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  24m 37s |  |  the patch passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  javac  |  24m 37s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  21m 50s |  |  the patch passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |  21m 50s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   3m 53s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   2m 27s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 29s |  |  the patch passed with JDK Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04  |
| +1 :green_heart: |  javadoc  |   1m 19s |  |  the patch passed with JDK Private Build-1.8.0_352-8u352-ga-1~20.04-b08  |
| +1 :green_heart: |  spotbugs  |   4m  8s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  26m 50s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  18m 10s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 30s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 54s |  |  The patch does not generate ASF License warnings.  |
|  |   | 250m 29s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5274 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux aa75e4c3c81a 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a4a2d2ed321d18d0d9c679fae3a193d8bccfb604 |
| Default Java | Private Build-1.8.0_352-8u352-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.17+8-post-Ubuntu-1ubuntu220.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_352-8u352-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/3/testReport/ |
| Max. process+thread count | 1239 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5274/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by sauraank** [comment]
For test4tests,
Not written a separate test for it, but manually tested logging levels of Error/info locally by throwing errors in case of both cases of async prefetch and sync get calls for read separately. Thanks.

**Pull request merged**